src/main/java/com/ait/lienzo/client/core/shape/Node.java

```diff
@@ -487,7 +486,7 @@ public void drawWithTransforms(final Context2D context, final double alpha, fina
         }
         if (context.isDrag() || isVisible())
         {
-            context.save();
+            context.saveContainer();
```

Comment:
I was expecting the call to `ctx.saveContainer()` at the `ContainerNode` class level instead of in the `Node` class... At least, before trying to understand the implementation details, that is what I was expecting in the look at the commit, but not sure if it makes sense, does it?

Reply:
@romartin if I insert the save/restore container call on the `ContainerNode` before the call to `drawWithTransforms`  it doesn't work properly, I thought about it and it does make sense IMO the call to be made on the `Node` since the node keeps the style,fill, stroke, information, WDYT?  
if you take a look at the javadoc on the `drawWithTransforms`

     * Used internally. Applies the node's transform-related attributes
     * to the current context, draws the node (and it's children, if any)
     * and restores the context.

Besides that I didn't want to change the current behavior, changing or inserting the method calls because on different places I'm not sure about the impact on Lienzo, since it does not have a good test coverage to guarantee nothing will be broken.

If we keep on this way, one option is to change the method name to saveNode, restoreNode, WDTY?

c/c @mdproctor

Reply:
yeah I was expecting the Node calling save/restore in order to populate the context with the attributes (fill, stroke, etc), and the ContainerNode calling the saveContainer/restoreContainer in order to serialize the children information. Anyway I have no detail on this so no worries if that doesn't work for now. Up to you, if you think the change is feasible and no risky we can do it, otherwise that's fine for me either.
THanks!